tests/functional/invalid_configs/test_invalid_where.py

```diff
@@ -0,0 +1,102 @@
+from struct import pack
+import os
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.fixtures import (
+    fact_orders_source_csv,
+    fact_orders_sql,
+    fact_orders_yml,
+)
+
+# models/invalid_where.sql
+invalid_where_sql = """
+select *
+from 
+{{ metrics.calculate(metric('invalid_where'), 
+    grain='month',
+    dimensions=['had_discount'],
+    where=["order_country='Japan'"]
+    )
+}}
+"""
+
+# models/invalid_where.yml
+invalid_where_yml = """
+version: 2 
+models:
+  - name: invalid_where
+    tests: 
+      - dbt_utils.equality:
+          compare_model: ref('invalid_where__expected')
+metrics:
+  - name: invalid_where
+    model: ref('fact_orders')
+    label: Total Discount ($)
+    timestamp: order_date
+    time_grains: [day, week, month]
+    type: sum
+    sql: order_total
+    dimensions:
+      - had_discount
+      - order_country
+"""
+
+class TestInvalidWhereMetric:
+
+    # configuration in dbt_project.yml
+    if os.getenv('dbt_target') == 'bigquery':
+        @pytest.fixture(scope="class")
+        def project_config_update(self):
+            return {
+            "name": "example",
+            "models": {"+materialized": "table"}
+            }
+    else: 
+        @pytest.fixture(scope="class")
+        def project_config_update(self):
+            return {
+            "name": "example",
+            "models": {"+materialized": "view"}
+            }  
+
+    # install current repo as package
+    @pytest.fixture(scope="class")
+    def packages(self):
+        return {
+            "packages": [
+                {"local": os.getcwd()}
+                ]
+        }
+
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "fact_orders_source.csv": fact_orders_source_csv
+            }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "fact_orders.sql": fact_orders_sql,
+            "fact_orders.yml": fact_orders_yml,
+            "invalid_where.sql": invalid_where_sql,
+            "invalid_where.yml": invalid_where_yml
+        }
+
+    def test_build_completion(self,project,):
+        # running deps to install package
+        results = run_dbt(["deps"])
+
+        # seed seeds
+        results = run_dbt(["seed"])
+        assert len(results) == 1
+
+        # Here we expect the run to fail because the incorrect
+        # time grain won't allow it to compile
```

Comment:
Is this comment a copy-paste mistake that's meant to say "because there's a where"?

Is this test to check that you managed to remove the old behaviour from everywhere?

Reply:
This is a copy paste mistake. Gonna remove those comments from everywhere because the tests describe themselves.

This test confirms that a column not used in the metric call cannot be included in the where clause